connection_helper.go

```diff
@@ -39,3 +42,12 @@ func (c *Connection) writeQuery(token int64, q []byte) error {
 
 	return c.writeData(data)
 }
+
+func (c *Connection) contextFromConnectionOpts() context.Context {
+	sum := c.opts.ReadTimeout + c.opts.WriteTimeout
```

Comment:
I am a little worried that this behaviour might be a little unexpected, I think it might make more sense to keep the previous behaviour if no context is passed to a query (I am just trying to think about existing code that might be affected)